### DISCUSSION TITLE
V8 initialState expanded should accept a function, not a boolean

### DISCUSSION THREAD

**Discussion by C-Higgins** [TanStack/table#4915] [Ideas]
As far as I can tell, there is no way to set the initial expanded rows conditionally. They are either all expanded or none. The interface of `initialState.expanded` should be the same as that of `getIsRowExpanded`.

**Comment by TonyVeigel** [comment]
Doesn't expanded take in an array:     
```
state: {
      expanded:[{1: true}],
    }
```

**Reply by C-Higgins** [reply]
It takes an object of rowID:boolean. You have to compute this manually outside the table, but you have no native access to table or column context while doing that, like you would if a function were accepted. You also have to force a table remount if the data is loaded async, as it won't re-apply the initial state after data loads. Not ergonomic.

**Reply by TonyVeigel** [reply]
I don't know I guess, I just use local component state to keep track of expanded rows initially, seems to work well to me. If you update the expanded row state that you send in to the table, the table updates as well. Same thing that the example shows.
